test/unit/math/prim/fun/zeros_int_array_test.cpp

```diff
@@ -0,0 +1,23 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+TEST(MathFunctions, zeros_int_array) {
+  using stan::math::zeros_int_array;
+
+  std::vector<int> u0 = zeros_int_array(0);
+  EXPECT_EQ(0, u0.size());
+
+  for (int K = 1; K < 5; K++) {
+    std::vector<int> u = zeros_int_array(K);
+    std::vector<int> v(K);
```

Comment:
`v` needs ot explicitly initialize to 0.  That's not guaranteed by the spec.

This also doesn't need a separate 0 case---it can get rolled into the loop.  At that point, you can save some vertical space by explicitly qualifing the call to `zeros_int_array`.